lib/assert.js

```diff
@@ -444,7 +444,48 @@ assert.doesNotThrow = function doesNotThrow(block, error, message) {
   throw actual;
 };
 
-assert.ifError = function ifError(err) { if (err) throw err; };
+assert.ifError = function ifError(err) {
+  if (err) {
+    let message = 'ifError got unwanted exception: ';
+    if (typeof err === 'object' && typeof err.message === 'string') {
+      if (err.message.length === 0 && err.constructor) {
+        message += err.constructor.name;
+      } else {
+        message += err.message;
+      }
+    } else {
+      message += inspect(err);
+    }
+
+    const newErr = new AssertionError({
+      actual: err,
+      expected: null,
+      operator: 'ifError',
+      message,
+      stackStartFn: ifError
+    });
+
+    // Make sure we actually have a stack trace!
+    const origStack = err.stack;
+
+    if (typeof origStack === 'string') {
+      const tmp2 = origStack.split('\n');
+      tmp2.shift();
+      // Filter all frames existing in err.stack.
+      let tmp1 = newErr.stack.split('\n');
+      for (var i = 0; i < tmp2.length; i++) {
+        const pos = tmp1.indexOf(tmp2[i]);
+        if (pos !== -1) {
+          tmp1 = tmp1.slice(0, pos);
+          break;
+        }
+      }
+      newErr.stack = `${tmp1.join('\n')}\n${tmp2.join('\n')}`;
+    }
```

Comment:
the goal of this code is not really clear, can you add some comments?

Reply:
Done.